### PR TITLE
Image snapshots

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- Implements support for Storybook "storyshots" + jest-image-snapshot
- Removes `All` stories (this pattern is duplicative of Storybook `addon-docs` functionality
- Enables developers to run storybook for a single package
- Added a story to design-tokens package to prove that package image-snapshot support is multi-packag
- **DOES NOT** enable CI support due to variance in snapshot generation between developer environment and Travis CI (haven't yet determined what's going wrong here).
  - For now developers should run `yarn image-snapshots` before submitting for PR

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] PR is ideally < ~400LOC

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,10 +14,11 @@ node_modules
 .eslintcache
 
 # code-gen
-packages/icons/src/generated
-www/public
 docs/static
+packages/icons/src/generated
 playground/dist
+storybook-static
```
Comment (ElliotTheHuman) [inline .gitignore:20]: Don't see a directory named `storybook-static`. But I assume this is correct?

File: packages/storybook-config/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Want to add image-snapshots to another package?
+
+1. Add .storybook to package root
+2. Add babel.config.js to package root (needed for Storybook to properly transpile TS)
+3. Add `storybook` and `storybook-build` scripts to package manifest (`package.json`)
+4. Copy `stories.shots.ts` to package `src/` directory (needs file-relative `path` to find stories and Storybook configuration)
+
+### TODO
+
+1. Add steps about registering package for Storybook composition at publish-time
```
Comment (ElliotTheHuman) [inline packages/storybook-config/README.md:16]: Wasn't sure if we wanted to keep this TODO in the `storybook-config/README`. If we do, then disregard.

File: config/deploy.sh [hunk 3]
```diff
@@ -2,10 +2,20 @@
...
 
 # Build Gatsby, then move contents to public folder for publishing
 yarn workspace www build
-mv www/public/* docs/static
+
+# Move everything into place...
+mv www/public docs/static
+
+# Move everything into place...
+mv storybook/build_static docs/static/storybook
```
Comment (ElliotTheHuman) [inline config/deploy.sh:17]: Not seeing `docs/static` when browsing our file system, but I imagine this is valid and there's magic afoot here that I don't fully understand when deploying.

**Review by ElliotTheHuman — APPROVED** [review]
Will be honest and say a lot of the configuration work goes over my head. That being said, from a pure "Does any of this code look broken?" stand point it all looks like valid TypeScript and JavaScript code to me 🤷 

Will say that I _did_ check the Button snapshots included in this PR and they all look good to go. Also double checked out `yarn storybook` on this branch and outside of a few fonts being **Times New Roman-y**, the stories all look good to go as well.

File: README.md [hunk 4]
```diff
@@ -61,6 +60,16 @@ We recommend using [Node Version Manager (NVM)](https://github.com/nvm-sh/nvm#in
...
 
+#### Storybook
+
+Storybook is a tool we use develop, demonstrate and document components in isolation.
+
+- **yarn storybook** starts master Storybook (includes all stories)
+- **yarn workspace [workspace-package-name] storybook** starts a storybook with just stories within the specified package.
+- **yarn image-snapshots** takes image-snapshots of all stories within packages/\*\*
+  - `preimage-snapshots` is a yarn task that runs `storybooks-build` task before capturing image snapshots.
+- **storybooks-build** builds all packages/\* storybooks for publishing and/or image-snapshot testing.
```
[the remark this answers is not in the document]
Comment (lukelooker) [reply README.md:71]: Yep / doh!

File: README.md [hunk 5]
```diff
@@ -61,6 +60,16 @@ We recommend using [Node Version Manager (NVM)](https://github.com/nvm-sh/nvm#in
 - **yarn clean** remove all build artifacts
 - **yarn test** runs Jest across all packages
 
+#### Storybook
+
+Storybook is a tool we use develop, demonstrate and document components in isolation.
```
[the remark this answers is not in the document]
Comment (lukelooker) [reply README.md:65]: FIxed.

File: packages/storybook-config/README.md (hunk 2 above)
Comment (lukelooker) [reply packages/storybook-config/README.md:16]: Yes, it's mostly a TODO for my own reference - in the near future as stories move into packages we should implement Storybook's new "composition feature.

File: config/deploy.sh (hunk 3 above)
Comment (lukelooker) [reply config/deploy.sh:17]: This happens as part of the documentation site deployment process. It's explicitly `gitignore`'d as we don't want to check-in the build artifacts.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lukelooker) [reply README.md:65]: I'm going to push README fixes into Tree snapshots PR so I can land this PR.

**Pull request merged**